### PR TITLE
Drop feature gate in descheduler support design proposal

### PR DESCRIPTION
**Pull request by fossedihelm**
The descheduler support design proposal specifies that a feature gate will be added.
The feature gate can be dropped, since no work is added other than to add annotations.
The feature gate drop also has the advantage of saving work to the cluster admins; with the feature gate, if a cluster admin wants to enable the descheduler, they have to install it and add the feature gate to KubeVirt.
Without the feature gate they only have to install it.

Also, this feature gate only enables a support for the descheduler, which should always be enabled, and there are no advantages to not have it.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```
/cc @acardace @Barakmor1 @iholder101 @jean-edouard @xpivarc

### CODE REVIEW

**Comment by iholder101** [comment]
From my perspective, the most important thing here is that the descheduler is not a Kubevirt feature, hence it shouldn't be guarded with a feature gate.

If we want to provide a way of avoiding these extra annotations, we can always provide a **configuration** for doing so, but using a feature gate for it is wrong IMO.

However, I currently don't see a reason to provide such a configuration as a few possibly redundant annotations shouldn't cause any harm that I can see.

/lgtm

**Comment by acardace** [comment]
/lgtm

+1 Totally agree with this.

**Comment by fabiand** [comment]
Agree with @iholder101 - I tend to say that we want a cluster level way to opt out of setting the annotations.
Main reason: Some people might not want these annotations.

**Comment by fabiand** [comment]
Actually @fossedihelm when are we adding the annotations? unconditionally or under certain conditions?

**Comment by iholder101** [comment]
> Agree with @iholder101 - I tend to say that we want a cluster level way to opt out of setting the annotations. Main reason: Some people might not want these annotations.

Thanks @fabiand!

Just to clarify: If we want this to be configurable, then a cluster-wide configuration is the way to go rather than a feature gate. Saying that, I'm not sure we need to add such a configuration.

We're talking about pod-level annotations, which are considered as an "implementation detail" of our VM/VMI objects.

Do you think there's a reason for users to opt-out of these annotations?

**Comment by fossedihelm** [comment]
> Actually @fossedihelm when are we adding the annotations? unconditionally or under certain conditions?

It depends on the annotation:
- `descheduler.alpha.kubernetes.io/request-evict-only` will be added for every virt-launcher pod
- `descheduler.alpha.kubernetes.io/eviction-in-progress` to the source virt-launcher pod only when an evacuation migration is triggered (not every migration).

IOW, from a practical point of view, we are going to add only the first one because the second one will be added to the pod that (eventually) will be `Completed`.

To me, a single annotation to the virt-launcher pod (not VM nor vmi) should not disturb users.

**Comment by fabiand** [comment]
Ok, thanks. Reasonable.

So, is `descheduler.alpha.kubernetes.io/evict` https://github.com/kubernetes-sigs/descheduler/blob/17af29afe418d56aff1954ee4ee0392e29b3fd5e/README.md?plain=1#L947
still required today as well?

File: design-proposals/descheduler-support.md [hunk 1]
```diff
@@ -30,21 +30,19 @@ The descheduler should be able to operate on `virt-launcher` pods.
...
 2. In cases where the VM will be migrated, eviction requests on `virt-launcher` pods should return 429 HTTP code with a special reason message.
-3. In order for the deschduler to understand that a `virt-launcher` pod is pending migration, it should be annotated with `descheduler.alpha.kubernetes.io/eviction-in-progress` so it will not try to evict it again.
+3. In order for the descheduler to understand that a `virt-launcher` pod is pending migration, it should be annotated with `descheduler.alpha.kubernetes.io/eviction-in-progress` so it will not try to evict it again.
 4. In case the migration failed, the `descheduler.alpha.kubernetes.io/eviction-in-progress` annotation will be removed from the source `virt-launcer` so the descheduler will understand that it could try evicting it again.
 
 ## Feature Gate
-A `DeschedulerSupport` feature gate will be introduced in order to enable this feature.
+No feature gate will be present, as no work will be done in KubeVirt. KubeVirt will only add annotations which, in the worst case, will have no added value.
```
Comment (orelmisan) [inline design-proposals/descheduler-support.md:45]:
```suggestion
A feature gate is not needed.
```

**Review by orelmisan — COMMENTED** [review]
Thank you @fossedihelm.
I support adding cluster-wide configuration to select whether the annotations will be added.
But it could be added last.

File: design-proposals/descheduler-support.md (hunk 1 above)
Comment (fossedihelm) [reply design-proposals/descheduler-support.md]: Done, thanks!

**Review by orelmisan — APPROVED** [review]

**Comment by fabiand** [comment]
/approve

according to all previous positive reviews.

**Pull request merged**